packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -29,18 +38,49 @@ export class QuickFileOpenService implements QuickOpenModel {
     }
 
     private wsRoot: FileStat | undefined;
+    private showIgnored: boolean = false;
+    private opened: boolean = false;
 
     isEnabled(): boolean {
         return this.wsRoot !== undefined;
     }
+    isOpened(): boolean {
+        return this.opened;
+    }
+    isShowIgnored(): boolean {
+        return this.showIgnored;
+    }
 
     open(): void {
-        this.quickOpenService.open(this, {
-            placeholder: 'file name to search',
-            fuzzyMatchLabel: true,
-            fuzzyMatchDescription: true,
-            fuzzySort: true
-        });
+        let placeholderText = "";
+        placeholderText = 'file name to search Press ' + this.getkeyCommand() + ' to toggle Show / Hide .gitignore files';
+        this.showIgnored = !this.isShowIgnored();
+        if (!this.isOpened()) {
+            this.opened = true;
+            this.quickOpenService.open(this, {
+                placeholder: placeholderText,
+                fuzzyMatchLabel: true,
+                fuzzyMatchDescription: true,
+                fuzzySort: true,
+                onClose: () => {
+                    this.showIgnored = false;
+                    this.opened = false;
+                }
+            });
+
+        } else {
+            this.quickOpenService.refresh();
+        }
+    }
+
+    private getkeyCommand(): string {
```

Comment:
Capital `K`.

If there is no keybinding, this function should return `undefined`.  If it returns undefined, the caller should not include the `press $keybinding again to show/hide...` since it's not relevant.

Reply:
Done